### PR TITLE
DOC: Update the DatetimeIndex.normalize docstring

### PR DESCRIPTION
**Pull request by techytushar**
Signed-off-by: Tushar Mittal <chiragmittal.mittal@gmail.com>

Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################## Docstring (pandas.DatetimeIndex.normalize) ##################
################################################################################

Return DatetimeIndex with times converted to midnight.

When using DatetimeIndex, the time can be converted to midnight i.e.
00:00:00. This is useful in cases, when the time does not matter.
Length is unaltered. The timezones are unaffected. If the frequency
is 'H','min' or 'S' it changes to None otherwise it remains 
unaffected.

Returns
-------
normalized : DatetimeIndex

See Also
--------
DatetimeIndex.floor : Floor the DatetimeIndex to the specified freq.
DatetimeIndex.ceil : Ceil the DatetimeIndex to the specified freq.
DatetimeIndex.round : Round the DatetimeIndex to the specified freq.

Examples
--------
>>> df = pd.DatetimeIndex(start='2014-08-01 10:00', freq='H',
...                       periods=3, tz='Asia/Calcutta')
>>> df
DatetimeIndex(['2014-08-01 10:00:00+05:30',
               '2014-08-01 11:00:00+05:30',
               '2014-08-01 12:00:00+05:30'],
                dtype='datetime64[ns, Asia/Calcutta]', freq='H')
>>> df.normalize()
DatetimeIndex(['2014-08-01 00:00:00+05:30',
               '2014-08-01 00:00:00+05:30',
               '2014-08-01 00:00:00+05:30'],
               dtype='datetime64[ns, Asia/Calcutta]', freq=None)

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DatetimeIndex.normalize" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

### CODE REVIEW

**Review by IHackPy — APPROVED** [review]
LGTM

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -1770,11 +1770,31 @@ def date(self):
 
     def normalize(self):
         """
-        Return DatetimeIndex with times to midnight. Length is unaltered
+        Return DatetimeIndex with times converted to midnight.
+
+        When using DatetimeIndex, the time can be converted to midnight i.e.
+        00:00:00. This is useful in cases, when the time does not matter in
```
Comment (jreback) [inline pandas/core/indexes/datetimes.py]: This useful in cases where the time does not matter (remove the in DataFrame). The length of the DatetimeIndex is unaltered.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -1770,11 +1770,31 @@ def date(self):
...
+        See Also
+        --------
+        DatetimeIndex.floor : Floor the DatetimeIndex to the specified freq.
+        DatetimeIndex.ceil : Ceil the DatetimeIndex to the specified freq.
+        DatetimeIndex.round : Round the DatetimeIndex to the specified freq.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'date':['2015-05-01 18:47:05.060000',
+        ...                            '2014-05-01 18:47:05.110000']})
```
Comment (jreback) [inline pandas/core/indexes/datetimes.py]: can you add another example with a tz.

File: pandas/core/indexes/datetimes.py (hunk 1 above)
Comment (jreback) [reply pandas/core/indexes/datetimes.py]: maybe mention this does not affect tz's

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by techytushar** [comment]
@jreback required changes done, please review :grinning:

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -1770,11 +1770,38 @@ def date(self):
 
     def normalize(self):
         """
-        Return DatetimeIndex with times to midnight. Length is unaltered
+        Return DatetimeIndex with times converted to midnight.
+
+        When using DatetimeIndex, the time can be converted to midnight i.e.
+        00:00:00. This is useful in cases, when the time does not matter.
+        Length is unaltered. The timezones are unaffected. If the frequency
```
Comment (jreback) [inline pandas/core/indexes/datetimes.py]: I would remove the last sentence. freq is actually re-inferred here, so this is misleading.

**Review by jreback — CHANGES_REQUESTED** [review]
small changes, otherwise lgtm.

File: pandas/core/indexes/datetimes.py [hunk 4]
```diff
@@ -1770,11 +1770,38 @@ def date(self):
 
     def normalize(self):
         """
-        Return DatetimeIndex with times to midnight. Length is unaltered
+        Return DatetimeIndex with times converted to midnight.
```
Comment (marielen) [inline pandas/core/indexes/datetimes.py]: Please, make a simpler description. I suggest `Convert times to midnight`.

**Review by marielen — CHANGES_REQUESTED** [review]

**Comment by techytushar** [comment]
done!

**Review by jreback — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]
Looks good! 

I made a few edits to make it more generic for both DatetimeIndex and Series.dt (they share the docstring)

**Pull request merged**